tests/view_tests/urls.py

```diff
@@ -104,3 +104,7 @@
     ),
     url(r'^render_no_template/$', views.render_no_template, name='render_no_template'),
 ]
+
+urlpatterns += [
```

Comment:
The URL & view are needed, because the `set_language` view in i18n uses `translate_url` (on line 46 of views/i18n.py). If the URL & view don't exist, the bug doesn't get triggered, since call to `reverse` (on line 178 of urls/base.py) would throw a `NoReverseMatch`, in which case `translate_url` would just return the original url variable it was passed on as an argument at `set_language`.
The buggy behaviour happens, when the URL & view **do exist**, since that's when the said `reverse` gets to modify the given url, and `reverse` assumes decoded URLs.